src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -87,11 +81,55 @@ public struct MetricPoint
 
         public double DoubleValue { get; internal set; }
 
-        public long[] BucketCounts { get; internal set; }
+        internal MetricPointStatus MetricPointStatus { get; private set; }
 
-        public double[] ExplicitBounds { get; internal set; }
+        public long[] GetBucketCounts()
+        {
+            if (this.aggType == AggregationType.Histogram)
+            {
+                return this.histogramMeasurements.AggregatedBucketCounts;
+            }
+            else
+            {
+                throw new InvalidOperationException($"{nameof(this.GetBucketCounts)} is not supported for this metric type.");
```

Comment:
Should we really throw here? 🤔

Reply:
If we do want to throw, `NotSupportedException` is probably more appropriate.

Reply:
I think we should throw here. If we don't there would be no way for exporters to know if they are working correctly as they would just get some default values which is not the actual metric value.